### PR TITLE
Added more config to onsQuestion

### PR DESCRIPTION
**Pull request by jrbarnes9**
### What is the context of this PR?
To ensure questions are presented more consistently, and spacing between elements is the same, I've made changes to `onsQuestion` so it now contains config for the common question components:
- Title
- Description
- Definition (collapsible)
- Guidance (include/exclude panel)
- Justification (collapsible)
- Submit button

The only flexibility in the type of component used for the answer.

Also... 
To prevent the need to always set the `button` parameters in the definition and justification elements, I've added default text for collapsible hide button and made the button mandatory in the component. If it has been deemed necessary it should be mandatory. I've also added default text for the button `contextSuffix` so it's never empty.

### How to review
- Check all question elements can accept the required content and that all applicable answer types can be used.
- Check spacing between elements is consistent.

### CODE REVIEW

**Review by boxadesign — DISMISSED** [review]
Appears to work

**Comment by jrbarnes9** [comment]
[Issue #824: Ensure consistent vertical spacing in questionnaire templates](https://trello.com/c/dBfruIos/558-issue-824-ensure-consistent-vertical-spacing-in-questionnaire-templates)

File: src/components/question/examples/question-fieldset/index.njk [hunk 1]
```diff
@@ -9,91 +9,91 @@ layout: none
...
+  "title": "Question example",
+  "breadcrumbs": {
+    "ariaLabel": 'Breadcrumbs',
+    "itemsList": [
+      {
+        "url": '#0',
+        "text": 'Previous'
+      }
+    ]
+  }
```
Comment (rmccar) [inline src/components/question/examples/question-fieldset/index.njk:21]:
```suggestion
  "title": "Question example",
  "breadcrumbs": {
      "ariaLabel": 'Breadcrumbs',
      "itemsList": [
          {
              "url": '#0',
              "text": 'Previous'
          }
      ]
  }
```
Bit of a nit but indentation is inconsistent

File: src/components/question/_macro.njk [hunk 2]
```diff
@@ -11,41 +11,76 @@
...
+            {% endif %}
+        {% endcall %}
+    {% endset %}
+
+    {% set questionGuidance %}
+        {% from "components/panel/_macro.njk" import onsPanel %}
+        {% call onsPanel({
+            "classes": "ons-u-mb-m"
+        }) %}
+            {% if params.guidance.content %}
```
Comment (rmccar) [inline src/components/question/_macro.njk:43]:
```suggestion
            {% if params.guidance.content is defined and params.guidance.content %}
```

File: src/components/question/_macro.njk [hunk 3]
```diff
@@ -11,41 +11,76 @@
...
+                {% if item.listLeadingLine %}
+                    <p>{{ item.listLeadingLine }}</h2>
+                {% endif %}
+                {% from "components/lists/_macro.njk" import onsList %}
+                {{
+                    onsList({
+                        "itemsList": item.itemsList
+                    })
+                }}
+            {% endfor -%}
```
Comment (rmccar) [inline src/components/question/_macro.njk:59]:
```suggestion
            {% if params.guidance.lists is defined and  params.guidance.lists %}
                {%- for item in params.guidance.lists -%}
                    {% if item.listHeading is defined and item.listHeading %}
                        <h2 class="ons-u-fs-r--b">{{ item.listHeading }}</h2>
                    {% endif %}
                    {% if item.listLeadingLine is defined and item.listLeadingLine %}
                        <p>{{ item.listLeadingLine }}</h2>
                    {% endif %}
                    {% from "components/lists/_macro.njk" import onsList %}
                    {{
                        onsList({
                            "itemsList": item.itemsList
                        })
                    }}
                {% endfor -%}
            {% endif -%}
```

Should probably check here if guidance.lists exists as its not mandatory and add is defined in a couple of places

**Review by rmccar — CHANGES_REQUESTED** [review]
Few small things

**Review by rmccar — APPROVED** [review]

**Review by boxadesign — APPROVED** [review]

**Pull request merged**